src/communication_interface.cc

```diff
@@ -219,10 +203,10 @@ void CommunicationInterface::PublishLcmAndPauseStatus() {
 
 void CommunicationInterface::PublishRobotStatus() {
   // Try to lock data to avoid read write collisions.
-  std::unique_lock<std::mutex> lock(robot_data_mutex_);
-  if (robot_data_.has_robot_data) {
-    drake::lcmt_iiwa_status franka_status =
-        utils::ConvertToLcmStatus(robot_data_);
+  if (std::unique_lock<std::mutex> lock {robot_data_mutex_};
+      robot_data_.has_robot_data) {
+    drake::lcmt_iiwa_status franka_status {
+        utils::ConvertToLcmStatus(robot_data_)};
```

Comment:
Should the lock be inside the if? Does that not restrict scope?

Reply:
it should. the point is to restrict scope because we want to unlock when done and before publishing lcm.